cetmix_tower_server/tests/test_server_template.py

```diff
@@ -199,3 +199,98 @@ def test_create_server_from_template_action(self):
             "new_value",
             "New variable values should be 'new_values'",
         )
+
+    def test_server_template_copy(self):
+        """
+        Test duplicating a Server Template with variable values and server logs
+        """
+
+        # A server template
+        server_template = self.server_template_sample
+
+        # Add variable values to the server template
+        self.VariableValues.create(
+            {
+                "variable_id": self.variable_version.id,
+                "server_template_id": server_template.id,
+                "value_char": "test",
+            }
+        )
+
+        # Create a command for the server log
+        command_for_log = self.Command.create(
+            {
+                "name": "Get system info",
+                "code": "uname -a",
+            }
+        )
+
+        # Add server logs to the template
+        self.ServerLog.create(
+            {
+                "name": "Log from server template",
+                "server_template_id": server_template.id,
+                "log_type": "command",
+                "command_id": command_for_log.id,
+            }
+        )
+
+        # Duplicate the server template
+        copied_template = server_template.copy()
+
+        # Ensure the new server template was created with a new ID
+        self.assertNotEqual(
+            copied_template.id,
+            server_template.id,
+            "Copied server template should have a different ID from the original",
+        )
+
+        # Check that the copied template has the same number of variable values
+        self.assertEqual(
+            len(copied_template.variable_value_ids),
+            len(server_template.variable_value_ids),
+            (
+                "Copied template should have the same "
+                "number of variable values as the original"
+            ),
+        )
+
+        # Ensure the first variable value in the copied template matches the original
+        original_variable_value = server_template.variable_value_ids
+        copied_variable_value = copied_template.variable_value_ids
+
+        self.assertEqual(
```

Comment:
This check only the value. However it doesn't check the if variable itself is copied correctly.

Reply:
Thanks,I'll fix it